src/server/server-board-controller.js

```diff
@@ -0,0 +1,225 @@
+    board.tasks = restTasks;
+
+    updateBoardFile();
+    return true;
+}
+
+const updateTask = (task, newProperties) => {
+    return {
+        ...task,
+        ...newProperties
+    }
+}
+
+const addTask = (columnID, task) => {
+    const board = getBoard();
+
+    const newTaskID = getNewTaskID();
+    const newTask = {
+        id: newTaskID,
+        ...task
+    };
+
+    board.tasks[newTaskID] = newTask;
+    board.columns[columnID].tasks.push(newTaskID);
+
+    updateBoardFile();
+
+    return newTask;
+};
+
+const addColumn = ({columnName}) => {
+    const board = getBoard();
+
+    const newColumnID = getNewColumnID();
+    const newColumn = {
+        id: newColumnID,
+        name: columnName,
+        tasks: []
+    };
+
+    board.columns[newColumnID] = newColumn;
+    board.columnsOrder.push(newColumnID);
+
+    updateBoardFile();
+
+    return newColumn;
+};
+
+const moveColumn = (columnID, {sourceIndex, destinationIndex}) => {
+    const board = getBoard();
+    let column = board.columns[columnID];
+
+    if (!column) throw new RequestError(404, `Column with id ${columnID} does not found.`);
+
+    board.columnsOrder.splice(sourceIndex, 1);
+    board.columnsOrder.splice(destinationIndex, 0, columnID);
+
+    updateBoardFile();
+    return true;
+}
+
+const deleteColumn = (columnID) => {
+    const board = getBoard();
+    let column = board.columns[columnID];
+
+    if (!column) throw new RequestError(404, `Column with id ${columnID} does not found.`);
+
+    board.columnsOrder = board.columnsOrder.filter(column => column !== columnID);
+    board.columns[columnID].tasks.forEach(task => delete board.tasks[task]);
+    delete board.columns[columnID];
+
+    updateBoardFile();
+    return true;
+}
+
+const changeColumnName = (columnID, {columnName}) => {
+    const board = getBoard();
+    let column = board.columns[columnID];
+
+    if (!column) throw new RequestError(404, `Column with id ${columnID} does not found.`);
+
+    board.columns[columnID].name = columnName;
+
+    updateBoardFile();
+    return true;
+}
+
+const changeTaskName = (taskID, {taskName}) => {
+    const board = getBoard();
+    let task = board.tasks[taskID];
+
+    if (!task) throw new RequestError(404, `Task with id ${taskID} does not found.`);
+
+    board.tasks[taskID].name = taskName;
+
+    updateBoardFile();
+    return true;
+}
+
+const updateBoardFile = () => {
+    const board = getBoard();
+
+    try {
+        updateFile(filePath, board);
+    } catch (e) {
+        throw new RequestError(500, 'Data updating error.');
+    }
+};
+
+const getNewTaskID = () => {
+    const board = getBoard();
+    const newID = board.metadata.lastTaskID + 1;
+
+    board.metadata.lastTaskID = newID;
+    updateBoardFile();
+
+    return `t${newID}`;
+}
+const getNewColumnID = () => {
+    const board = getBoard();
+    const newID = board.metadata.lastColumnID + 1;
+
+    board.metadata.lastColumnID = newID;
+    updateBoardFile();
```

Comment:
nit: I would create a wrapper for the functions working with file that will execute the function, take returned value, then execute `updateBoardFile` and return the value from the function. And I would use this wrapper with all the functions in the exported by default object

Reply:
@alex-zh WDYT? Anna didn't like that I call updateBoardFile() in every board-controller file but we couldn't refactor it in correct way.